src/plugins/intel_cpu/src/nodes/kernels/stack_allocator.hpp

```diff
@@ -0,0 +1,477 @@
+            IE_THROW() << "Cannot allocate Address out of transaction. Please, finish first transaction !!";
+        }
+        stack_allocator_.commit();
+    }
+
+    Address(const Address& addr) = delete;
+    Address& operator=(const Address& addr) = delete;
+    Address(Address&& addr) noexcept = delete;
+    Address& operator=(Address&& addr) noexcept = delete;
+
+    virtual ~Address() {
+        if (transaction_) {
+            release(*transaction_);
+        } else {
+            release();
+            stack_allocator_.commit();
+        }
+    }
+
+    void release(Transaction& transaction) {
+        if (allocation_) {
+            transaction.begin();
+            allocation_->is_used = false;
+            stack_allocator_.deallocate();
+        }
+        allocation_ = {};
+    }
+
+    void release() {
+        if (allocation_) {
+            if (stack_allocator_.isTransaction()) {
+                IE_THROW() << "Cannot release Address out of transaction. Please, finish first transaction !!";
+            }
+            allocation_->is_used = false;
+            stack_allocator_.deallocate();
+            stack_allocator_.commit();
+        }
+        allocation_ = {};
+    }
+
+    operator Xbyak::Address&() {
+        ensureValid();
+        return allocation_->address;
+    }
+
+    operator const Xbyak::Address&() const {
+        ensureValid();
+        return allocation_->address;
+    }
+
+    virtual Address& operator=(const Xbyak::Xmm& vmm) {
+        stack_mov(*this, vmm);
+        return *this;
+    }
+
+    virtual Address& operator=(const Xbyak::Reg& reg) {
+        stack_mov(*this, reg);
+        return *this;
+    }
+
+    explicit operator bool() const {
+        return isInitialized();
+    }
+
+    bool isInitialized() const {
+        return allocation_ && !allocation_->is_transaction;
+    }
+
+    friend void ::ov::intel_cpu::stack_mov(Address& addr, const Xbyak::Xmm& vmm);
+    friend void ::ov::intel_cpu::stack_mov(Address& addr, const Xbyak::Reg& reg);
+    friend void ::ov::intel_cpu::stack_mov(const Xbyak::Xmm& vmm, const Address& addr);
+    friend void ::ov::intel_cpu::stack_mov(const Xbyak::Reg& reg, const Address& addr);
+
+private:
+    void ensureSize(const Xbyak::Reg& reg) const {
+        ensureValid();
+        const size_t reg_size = reg.getBit() / 8;
+        if (reg_size > allocation_->size) {
+            IE_THROW() << "reg size is bigger than space allocated in StackAllocator !!";
+        }
+    }
+
+    void ensureValid() const {
+        if (!isInitialized()) {
+            IE_THROW() << "StackAllocator::Address is either not initialized or released !!";
+        }
+    }
+
+    x64::jit_generator& generator() const {
+        return stack_allocator_.code_generator;
+    }
+
+    Transaction* transaction_{};
+    StackAllocator& stack_allocator_;
+    Allocation::Ptr allocation_;
+};
+
+template<typename TReg>
+class StackAllocator::Reg : public StackAllocator::Address {
```

Comment:
The class name is slightly misleading. Actually we cannot use the objects of this type as registers. If we had modern C++ standard in use we could just add templated constructors to `Adress` class, but it is what it is, we need a separate templated class. M.b. `StackAllocator::RegAdress` is better name? What do you think?

Reply:
`StackAllocator::RegAddress` I thought about it, but as for me `StackAllocator::Reg` already have good enough prefix `StackAllocator::` which describe where memory for register could be allocated.
Mostly I choose the `StackAllocator::Reg`, because `StackAllocator::RegAddress` is too long as for me and that little bit decrease readability

Reply:
M.b., but it really confusing, at least for me as a person who got a fresh look on this code. Thus, it probably means that other people who are not familiar with this class will have the same impression.

Reply:
Renamed to RegAddress. Please, see https://github.com/openvinotoolkit/openvino/pull/13790